### PR TITLE
build(aio): rendering of optional parameters

### LINKED ISSUE #22134: aio + docs/api: indicate when a param is optional
This is information that dgeni has and should be rendered in the template.

### PR DESCRIPTION
**Pull request by petebacondarwin**
Closes #22134 

Now we automatically render `Optional.` at the start of the parameter description if the parameter is not required. We also render the value given to the parameter if it is not provided in the call.

An example is in the Location class... (https://pr22435-d01479e.ngbuilds.io/api/common/Location#path)

screen shot 2018-02-25 at 09 24 44

Note that if there is no description for the parameter then we currently just display the type of the parameter. This is up for negotiation.

### CODE REVIEW

File: aio/tools/transforms/templates/api/lib/paramList.html [hunk 1]
```diff
@@ -19,9 +19,13 @@
     <tr class="{$ parameterClass $}">
       <td class="param-name"><a id="{$ parameter.anchor $}"></a>{$ parameter.name $}</td>
       <td class="param-description">
-        {% if parameter.description | trim %}{$ parameter.description | marked $}
+      {% marked %}
+        {% if parameter.isOptional or parameter.defaultValue !== undefined %}Optional. Default is `{$ parameter.defaultValue === undefined and 'undefined' or parameter.defaultValue $}`.{% endif %}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/lib/paramList.html]: I would consider placing this after the description/type.

**Review by gkalpak — APPROVED** [review]

File: aio/tools/transforms/templates/api/lib/paramList.html (hunk 1 above)
Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/lib/paramList.html]: I was following the example given in @jbogarthyde's Google doc "Animations API example".

screen shot 2018-02-25 at 20 55 32

Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/lib/paramList.html]: We can change the ordering around later, as we get more eyes on this - especially when parameters actually start getting descriptions :-P . Meanwhile the question of whether/where to show the type of the param is more problematic IMO.

Comment (gkalpak) [reply aio/tools/transforms/templates/api/lib/paramList.html]: Normally, I would suggest having the type in a separate column. But given the fact that some types will be too verbose and that the types are already shown in the signature, I think it is fine to just show the description.

**Pull request closed without merging**